Framework/PythonInterface/plugins/algorithms/PEARLTransfit.py

```diff
@@ -0,0 +1,370 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+# -----------------------------------------------------------------------
+# Transfit v2 - translated from Fortran77 to Python/Mantid
+# Original Author: Christopher J Ridley
+# Last updated: 2nd October 2020
+#
+# Program fits a Voigt function to PEARL transmission data, using the doppler broadening
+# of the gaussian component to determine the sample temperature
+#
+#
+# References:
+# 'Temperature measurement in a Paris-Edinburgh cell by neutron resonance spectroscopy' - Journal Of Applied Physics 98,
+# 064905 (2005)
+# 'Remote determination of sample temperature by neutron resonance spectroscopy' - Nuclear Instruments and Methods in
+# Physics Research A 547 (2005) 601-615
+# -----------------------------------------------------------------------
+from mantid.kernel import Direction
+from mantid.api import PythonAlgorithm, MultipleFileProperty, AlgorithmFactory, mtd
+from mantid.simpleapi import (LoadRaw, DeleteWorkspace, CropWorkspace, NormaliseByCurrent, ExtractSingleSpectrum,
+                              ConvertUnits, CreateWorkspace, Fit, Plus, Multiply, RenameWorkspace,
+                              CreateSingleValuedWorkspace)
+from os import path
+import numpy as np
+
+
+class PEARLTransfit(PythonAlgorithm):
+
+    def version(self):
+        return 1
+
+    def name(self):
+        return 'PEARLTransfit'
+
+    def category(self):
+        return 'PEARL tools'
+
+    def summary(self):
+        return "Reads high-energy neutron resonances from the downstream monitor data on the PEARL instrument," \
+               "then fits a Voigt function to them to determine the sample temperature. A calibration must be run for" \
+               "each sample temperature. Can be used on a single file, or multiple files, in which case workspaces" \
+               "are summed and the average taken."
+
+    def PyInit(self):
+        self.declareProperty(MultipleFileProperty('Files', extensions=[".raw", ".s0x"]),
+                             doc='Files of calibration runs (numors). Must be detector scans.')
+        self.declareProperty(name='FoilType',
```

Comment:
This FoilType parameter feels like it could usefully be a drop down.
You can make this happen by adding a parameter called validator like:
validator=StringListValidator(['Value1','Value2'])